node/worker/worker.go

```diff
@@ -93,9 +90,8 @@ func (w *Worker) SelectTransactionsForNewBlock(newBlockNum uint64, txs types.Tra
 	unselected := types.Transactions{}
 	invalid := types.Transactions{}
 	for _, tx := range txs {
-		if tx.ShardID() != w.shardID && tx.ToShardID() != w.shardID {
+		if tx.ShardID() != w.shardID {
 			invalid = append(invalid, tx)
-			continue
```

Comment:
this is a bug fix. this is causing stress testing to fail on s3 @harmony-ek  @LeoHChen

Reply:
Thanks for spotting this and also opening #1471 to fix it.  I *knew* something was amiss here… :-p